web-common/src/features/chat/core/messages/CallMessage.svelte

```diff
@@ -0,0 +1,207 @@
+  import type { V1Message } from "../../../../runtime-client";
+  import { isHiddenAgentTool, parseChartData } from "../utils";
+  import ChartBlock from "./ChartBlock.svelte";
+
+  export let message: V1Message;
+  export let resultMessage: V1Message | undefined = undefined;
+
+  let isExpanded = false;
+
+  // Call message properties
+  $: toolName = message.tool || "Unknown Tool";
+  $: toolInput = formatContentData(message);
+  $: isHidden = isHiddenAgentTool(message.tool);
+
+  // Result message properties
+  $: hasResult = !!resultMessage;
+  $: isError = resultMessage?.contentType === "error";
+  $: resultContent = resultMessage ? formatContentData(resultMessage) : "";
+
+  // Chart detection and parsing
+  $: isChart = isChartCall(message);
+  $: chartData = isChart
+    ? parseChartData({ input: message.contentData })
+    : null;
+
+  function toggleExpanded() {
+    isExpanded = !isExpanded;
+  }
+
+  function formatContentData(msg: V1Message): string {
+    const rawContent = msg.contentData || "";
+
+    switch (msg.contentType) {
+      case "json":
+        try {
+          const obj = JSON.parse(rawContent);
+          return JSON.stringify(obj, null, 2);
+        } catch {
+          return rawContent;
+        }
+
+      case "text":
+        return rawContent;
+
+      case "error":
+        return rawContent;
+
+      default:
+        // Fallback for unknown content types
+        return rawContent;
+    }
+  }
+
+  function isChartCall(message: V1Message): boolean {
+    return message.tool === "create_chart";
+  }
+</script>
+
+{#if !isHidden}
+  <div class="tool-container">
+    <button class="tool-header" on:click={toggleExpanded}>
+      <div class="tool-icon">
+        {#if isExpanded}
+          <CaretDownIcon size="16" />
+        {:else}
+          <ChevronRight size="16" />
+        {/if}
+      </div>
+      <div class="tool-name">
+        {toolName}
+      </div>
+    </button>
+
+    {#if isExpanded}
+      <div class="tool-content">
+        <div class="tool-section">
+          <div class="tool-section-title">Request</div>
+          <div class="tool-section-content">
+            <pre class="tool-json">{toolInput}</pre>
+          </div>
+        </div>
+
+        {#if hasResult}
+          <div class="tool-section">
+            <div class="tool-section-title">
+              {isError ? "Error" : "Response"}
+            </div>
+            <div class="tool-section-content">
+              <pre class="tool-json">{resultContent}</pre>
+            </div>
+          </div>
+        {/if}
+      </div>
+    {/if}
+  </div>
+
+  {#if isChart && chartData && hasResult && !isError}
+    <!-- Chart visualization (shown below the collapsible tool details) -->
+    <div class="chart-container">
+      <ChartBlock
+        chartType={chartData.chartType}
+        chartSpec={chartData.chartSpec}
+      />
+    </div>
+  {/if}
+{/if}
+
+<style lang="postcss">
+  .tool-container {
+    border: 1px solid #e5e7eb;
```

Comment:
This is in a lot places in AI components. We should use this as an opportunity to avoid hardcoded colors. Also use tailwind for all these styles.